File: est-mono/src/main/java/ru/est/mono/service/BoardService.java
package ru.est.mono.service;

import ru.est.mono.model.BoardDto;
import ru.est.mono.model.BoardListDto;
import ru.est.mono.model.CreateRequest;
import ru.est.mono.model.ShareBoardDto;
import ru.est.mono.model.UnshareRequest;

import java.util.UUID;

public interface BoardService {

    BoardListDto getBoards();

    BoardDto getByUuid(UUID boardId);

    BoardDto create(CreateRequest createRequest);

    BoardDto update(UUID boardId, CreateRequest createRequest);

    void delete(UUID boardId);

    void share(UUID boardId, ShareBoardDto shareBoardDto);

    void unshare(UUID boardId, UnshareRequest unshareRequest);

    void changeAccess(UUID boardId, ShareBoardDto shareBoardDto);
}

File: est-proxy/src/service/board_service.go
package service

import (
	"context"
	"est-proxy/src/errors"
	proxymodels "est_proxy_go/models"
	"github.com/google/uuid"
)

type BoardService interface {
	GetByUuid(ctx context.Context, userId *uuid.UUID, boardId *uuid.UUID) (*proxymodels.BoardDto, *errors.StatusError)
	List(ctx context.Context, userId *uuid.UUID) (*proxymodels.BoardListDto, *errors.StatusError)
	Create(ctx context.Context, userId *uuid.UUID, createRequest *proxymodels.CreateRequest) (*proxymodels.BoardDto, *errors.StatusError)
	Update(ctx context.Context, userId *uuid.UUID, boardId *uuid.UUID, updateRequest *proxymodels.CreateRequest) (*proxymodels.BoardDto, *errors.StatusError)
	DeleteBoard(ctx context.Context, userId *uuid.UUID, boardId *uuid.UUID) *errors.StatusError
	Share(ctx context.Context, userId *uuid.UUID, boardId *uuid.UUID, shareBoardDto *proxymodels.ShareBoardDto) *errors.StatusError
	ChangeAccess(ctx context.Context, userId *uuid.UUID, boardId *uuid.UUID, shareBoardDto *proxymodels.ShareBoardDto) *errors.StatusError
	Unshare(ctx context.Context, userId *uuid.UUID, boardId *uuid.UUID, unshareRequest *proxymodels.UnshareRequest) *errors.StatusError
}
